core/src/main/java/com/linecorp/armeria/common/util/SystemInfo.java

```diff
@@ -154,8 +162,20 @@ public static long currentTimeMicros() {
         return JavaVersionSpecific.get().currentTimeMicros();
     }
 
-    private static boolean isLinux() {
-        return Ascii.toLowerCase(System.getProperty("os.name", "")).startsWith("linux");
+    /**
+     * Returns the operating system type.
+     */
+    public static OsType osType() {
+        return osType;
+    }
+
+    public enum OsType {
+        WINDOWS,
+        LINUX,
+        MAC,
+        FREEBSD,
```

Comment:
I would lean towards leaving out `FREEBSD` and `SOLARIS` - it's unlikely Armeria servers would have logic specific to them anyways so they can go in `OTHERS`

Reply:
OK. 😄 
I thought FREEBSD and SOLARIS as OTHERS and changed it.